HoloPipelines/pipelines/components/compNumpyTransformation.py

```diff
@@ -0,0 +1,47 @@
+import scipy.ndimage
+
+
+def crop(img, cropx, cropy, cropz):
+    x, y, z = img.shape
+
+    startX = x // 2 - (cropx // 2)
+    startY = y // 2 - (cropy // 2)
+    startZ = z // 2 - (cropz // 2)
+
+    endX = startX + cropx
+    endY = startY + cropy
+    endZ = startZ + cropz
+
+    return img[startX:endX, startY:endY, startZ:endZ]
+
+
+def resize(img, ratioX, ratioY, ratioZ):
+    return scipy.ndimage.interpolation.zoom(img, [ratioX, ratioY, ratioZ])
+
+
+def sizeLimit(img, limit):
+    if len(img.shape) == 5:
+        x, y, z, t1, t2 = img.shape
+    else:
```

Comment:
making an assumption that "img" will be of size 5 or 3. Perhaps handle other incorrect usages of this function.

Reply:
update to:
- check if len >= 3
- read the first 3 value from `img.shape`